### PR TITLE
github: Fix permissions so they are minimal

### PR DESCRIPTION
**Pull request by illume**
See 
- https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs
- https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions

Note: `contents: read` at the top turns off all other permissions. It's best practice to do that and specify in each job any extra permissions they have. This is so that it's easier to tell which permission is for which job, and also so that the other permissions are turned off or not.

### CODE REVIEW

File: .github/workflows/container-publish.yml [hunk 1]
```diff
@@ -21,7 +24,6 @@ jobs:
     name: Test building container image
     runs-on: ubuntu-latest
     permissions:
-      contents: read
       packages: write
```
Comment (yolossn) [inline .github/workflows/container-publish.yml:27]:
```suggestion
      packages: write  # needed for publishing the container image.
```

File: .github/workflows/docker-extension-release.yml [hunk 2]
```diff
@@ -6,13 +6,18 @@ on:
...
+
 env:
   REGISTRY: docker.io
   IMAGE_NAME: headlamp/headlamp-docker-extension
 jobs:
   build_and_push_docker_extension:
     name: Build docker extension
     runs-on: ubuntu-latest
+    permissions:
+      secrets: read
```
Comment (yolossn) [inline .github/workflows/docker-extension-release.yml:20]:
```suggestion
      secrets: read  # needed to fetch docker hub creds
```

File: .github/workflows/draft-release.yml [hunk 3]
```diff
@@ -8,17 +8,21 @@ on:
...
         default: 'X.Y.Z'
 
+permissions:
+  contents: read
+
 jobs:
   build:
     permissions:
       contents: write
+      secrets: read
```
Comment (yolossn) [inline .github/workflows/draft-release.yml:18]: since we are using `github.token` permission to read secrets isn't required.

File: .github/workflows/helm-chart-release.yml [hunk 4]
```diff
@@ -26,14 +32,12 @@ jobs:
...
       - name: Run chart-releaser
-        uses: helm/chart-releaser-action@v1.2.1
+        uses: helm/chart-releaser-action@v1.6.0
         env:
-          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          CR_TOKEN: "${{ github.token }}"
         with:
           config: .github/cr.yaml
-
+          mark_as_latest: false # only headlamp is set to latest
```
Comment (yolossn) [inline .github/workflows/helm-chart-release.yml:43]: good addition.

File: .github/workflows/trigger-flatpak-update.yml [hunk 5]
```diff
@@ -7,10 +7,17 @@ on:
...
+permissions:
+  contents: read
+
 jobs:
   trigger_flatpak_update:
     name: Trigger Flatpak Headlamp version update
     runs-on: ubuntu-latest
+
+    permissions:
+      contents: secrets
```
Comment (yolossn) [inline .github/workflows/trigger-flatpak-update.yml:19]:
```suggestion
      contents: read
      secrets: read
```

File: plugins/examples/pod-counter-old/.gitignore
Comment (yolossn) [inline plugins/examples/pod-counter-old/.gitignore:1]: The changes related to renaming the pod-counter plugin folder name can be part of another PR.

**Review by yolossn — CHANGES_REQUESTED** [review]

File: .github/workflows/draft-release.yml (hunk 3 above)
Comment (illume) [reply .github/workflows/draft-release.yml]: Thanks, fixed.

File: .github/workflows/trigger-flatpak-update.yml (hunk 5 above)
Comment (illume) [reply .github/workflows/trigger-flatpak-update.yml]: Thanks, fixed.

File: plugins/examples/pod-counter-old/.gitignore
Comment (illume) [reply plugins/examples/pod-counter-old/.gitignore:1]: Thanks, fixed.

**Review by illume — COMMENTED** [review]
Thanks @yolossn, I've applied your fixes.

**Review by skoeva — COMMENTED** [review]
the github context for the token seems more appropriate, overall looks good!

**Review by knrt10 — APPROVED** [review]
Looks good

**Review by joaquimrocha — APPROVED** [review]
